src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs

```diff
@@ -89,6 +89,39 @@ public async Task<Response<ConfigurationSetting>> AddAsync(ConfigurationSetting
             }
         }
 
+        public async Task<Response<ConfigurationSetting>> AddAsync(string key, string value, CancellationToken cancellation = default)
+        {
+            if (string.IsNullOrEmpty(key)) throw new ArgumentNullException($"{nameof(key)}");
+            if (string.IsNullOrEmpty(value)) throw new ArgumentNullException($"{nameof(value)}");
+
+            Uri uri = BuildUriForKvRoute(key);
+
+            using (HttpMessage message = _pipeline.CreateMessage(cancellation))
+            {
+                ReadOnlyMemory<byte> content = Serialize(value);
+                message.SetRequestLine(HttpVerb.Put, uri);
+
+                message.AddHeader("Host", uri.Host);
+                message.AddHeader(IfNoneMatch, "*");
+                message.AddHeader(MediaTypeKeyValueApplicationHeader);
+                message.AddHeader(HttpHeader.Common.JsonContentType);
+                message.AddHeader(HttpHeader.Common.CreateContentLength(content.Length));
+                AddClientRequestID(message);
+                AddAuthenticationHeaders(message, uri, HttpVerb.Put, content, _secret, _credential);
+
+                message.SetContent(HttpMessageContent.Create(content));
+
+                await _pipeline.SendMessageAsync(message).ConfigureAwait(false);
+
+                var response = message.Response;
+                if (response.Status == 200)
```

Comment:
I would check for 200 or 201 (the service is supposed to return 201/Created if the resource was actually created. I don't think it currently does so, but adding 201 for success should not harm and would make the library more forward compatible).